qemu/tests/mlock_basic.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from subprocess import check_output
+
+from virttest import env_process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    [Mlock] Basic test, this case will:
+    1) Get nr_mlock and nr_unevictable in host before VM start.
+    2) Start the VM.
+    3) Get nr_mlock and nr_unevictable in host after VM start.
+    4) Check nr_mlock and nr_unevictable with VM memory.
+    5) Check kernel crash
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_mlock_unevictable(mlock_cmd, unevictable_cmd):
+        """
+        Get nr_mlock and nr_unevictable in host
+
+        :param mlock_cmd: CMD to get nr_mlock
+        :param unevictable_cmd: CMD to get nr_unevictable
+        """
+        mlock = check_output(mlock_cmd).strip().split().pop()
+        unevictable = check_output(unevictable_cmd).strip().split().pop()
+        return mlock, unevictable
+
+    mlock_cmd = params["mlock_cmd"].split()
+    unevictable_cmd = params["unevictable_cmd"].split()
+    vm_mem = params["mem"]
+
+    error_context.context("Get nr_mlock and nr_unevictable in host before VM start!", logging.info)
+    mlock_pre, unevictable_pre = get_mlock_unevictable(mlock_cmd, unevictable_cmd)
+    error_context.context("mlock_pre is %s and unevictable_pre is %s." %
+                          (mlock_pre, unevictable_pre), logging.info)
+    params["start_vm"] = "yes"
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
+
+    error_context.context("Get the main VM!", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    error_context.context("Get nr_mlock and nr_unevictable in host after VM start!", logging.info)
+    mlock_post, unevictable_post = get_mlock_unevictable(mlock_cmd, unevictable_cmd)
+    error_context.context("mlock_post is %s and unevictable_post is %s." %
+                          (mlock_post, unevictable_post), logging.info)
+
+    error_context.context("Check nr_mlock and nr_unevictable with VM memory!")
+    realtime_mlock = params["realtime_mlock"]
+    if realtime_mlock == "on":
+        nr_mlock = int(mlock_post) - int(mlock_pre)
+        if nr_mlock < int(vm_mem)*1024/4:
+            error_context.context("nr_mlock = %d, vm_mem = %s." % (nr_mlock, vm_mem), logging.info)
+            test.fail("nr_mlock is not fit with VM memory when mlock is %s!" % realtime_mlock)
+        nr_unevictable = int(unevictable_post) - int(unevictable_pre)
+        if nr_unevictable < int(vm_mem)*1024/4:
```

Comment:
The memory page size is 4K, so the number of pages is equal to memory size in mega multiplied by 1024 and divided by 4. Thanks!